docs/architecture.md

```diff
@@ -1,14 +1,102 @@
-### cozy-client architecture
+### Cozy-client Architecture
 
 This is an overview of the cozy-client architecture:
 
 ![Architecture schema](./architecture.png)
 
-The architecture schema can also be accessed [online here](https://whimsical.co/6AJnnS7v5ePRcuKZysPAF5).
-It's created on https://whimsical.co/ — if you need to edit it, ask for an invite from one of the projects maintainers.
+<!-- The architecture schema can also be accessed [online here](https://whimsical.co/6AJnnS7v5ePRcuKZysPAF5).
+It's created on https://whimsical.co/ — if you need to edit it, ask for an invite from one of the projects maintainers. -->
 
-### Missing parts
+To explain the architecture, we will see how a request for data flows
+through cozy-client and how finally the UI is updated.
 
-- Init
-- Associations
-- Mutations
+### Query defitions
+
+To query data, we create a *query definition*: an object describing documents to fetch.
+
+```
+{
+  doctype: "io.cozy.todos",
+  selector: {
+    "finished": true
+  }
+}
+```
+
+It is typically created via the helper `Q` that provides a fluid interface
+to create a *Query Definition*.
+
+```jsx
+const qdef = Q('io.cozy.todos').where({ finished: true })
+```
+
+Then we need to execute this query, fetch the data, and storing it.
+
+```jsx
+await client.query(qdef)
+```
+
+### Links
+
+When executed, *Query definitions* are passed down to *Links*.
+
+Links accept query definitions and can choose
+
+- either to return a response
+- or to pass the query definition down to the next link. 
+
+ℹ️ This is how offline support is implemented : on mobile, a PouchLink is added as a link to cozy-client, before the default StackClientLink. The 
```

Comment:
```suggestion
ℹ️ This is how offline support is implemented: 
on mobile, a PouchLink (we use PouchDB under the wood)  is added as a link to cozy-client, before the default StackLink (`stack link` because cozy's API backend is called `cozy-stack`.) The 
```